src/image/PhotoCalib.cc

```diff
@@ -394,25 +394,29 @@ void PhotoCalib::instFluxToMagnitudeArray(afw::table::SourceCatalog const &sourc
     }
 }
 
-std::shared_ptr<PhotoCalib> makePhotoCalib(daf::base::PropertySet &metadata, bool strip) {
-    double calibration = 0.0, calibrationErr = 0.0;
+std::shared_ptr<PhotoCalib> makePhotoCalibFromMetadata(daf::base::PropertySet &metadata, bool strip) {
+    double instFluxMag0 = 0.0, instFluxMag0Err = 0.0;
 
     auto key = "FLUXMAG0";
     if (metadata.exists(key)) {
-        double fluxMag0 = metadata.getAsDouble(key);
+        instFluxMag0 = metadata.getAsDouble(key);
         if (strip) metadata.remove(key);
 
-        calibration = utils::referenceFlux / fluxMag0;
         key = "FLUXMAG0ERR";
         if (metadata.exists(key)) {
-            double fluxMag0Err = metadata.getAsDouble(key);
-            calibrationErr = utils::referenceFlux * fluxMag0Err / std::pow(fluxMag0, 2);
+            instFluxMag0Err = metadata.getAsDouble(key);
             if (strip) metadata.remove(key);
         }
     } else {
         return nullptr;
     }
 
+    return makePhotoCalibFromCalibZeroPoint(instFluxMag0, instFluxMag0Err);
```

Comment:
I was kind of hoping you'd move this inside the `if` statement... as it is, you've got code that's executed only if `"FLUXMAG0"` exists, followed by code that's executed only if it doesn't exist, followed by code that's executed only if it does exist. :spaghetti: As it is, it's not at all obvious that the default value of `instFluxMag0` never gets used.

Reply:
Good point. How is it now?